natss/pkg/reconciler/controller/controller_test.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"testing"
+
+	"knative.dev/pkg/injection/sharedmain"
+
+	"knative.dev/pkg/injection"
```

Comment:
you just need to import this:

```go
	_ "knative.dev/pkg/client/injection/kube/informers/apps/v1/deployment/fake"
	_ "knative.dev/eventing-contrib/natss/pkg/client/injection/informers/messaging/v1alpha1/natsschannel/fake"
	_ "knative.dev/pkg/client/injection/kube/informers/core/v1/endpoints/fake"
	_ "knative.dev/pkg/client/injection/kube/informers/core/v1/service/fake"
	_ "knative.dev/eventing-contrib/natss/pkg/client/injection/informers/messaging/v1alpha1/natsschannel/fake"
```

and that should work

this loads up the fakes into the context that NewController is pulling the resources from.

i.e., from these calls in `NewController`: 

```go
	channelInformer := natsschannel.Get(ctx)
	deploymentInformer := deploymentinformer.Get(ctx)
	serviceInformer := service.Get(ctx)
	endpointsInformer := endpoints.Get(ctx)
	kubeClient := kubeclient.Get(ctx)
```

This is special Knative injection: https://github.com/knative/pkg/tree/master/injection

Reply:
then should I use `ctx, _ := injection.Fake.SetupInformers(context.Background(), &rest.Config{})` for getting the context?
The test on my machine doesn't fail :confused:

Reply:
@n3wscott Thank you! :smile: